### ISSUE TITLE
[conn] Failed to resolve new client Error

### ISSUE THREAD

**Issue by kidroca** [facebook/flipper#3430]
## 🐛 Bug Report

Since updating I'm unable to see my app in Flipper
I see some connection errors
Reverting back to the previous version fixes the issue for me

![image](https://user-images.githubusercontent.com/12156624/153474939-a464ed2e-57fc-408b-8a20-2a68568625ad.png)

```
8 plugin(s) were loaded in legacy mode. Please visit https://fbflipper.com/docs/extending/sandy-migration to learn how to migrate these plugins to the new Sandy architecture: 
Hermes Debugger (RN) (id: Hermesdebuggerrn)
KaiOS RAM graph (id: kaios-graphs)
KaiOS: big allocations (id: kaios-big-allocations)
Layout (id: Inspector)
LeakCanary (id: LeakCanary)
Performance (id: flipper-plugin-performance)
React Native Tic Tac Toe (id: ReactNativeTicTacToe)
Sandbox (id: Sandbox) 
Failed to initialize ADB. Please disable Android support in settings, or configure a correct path. 
Error: spawn D:\Android\SDK\adb ENOENT
[conn] RSockets are being deprecated at Flipper. Please, migrate to WebSockets. App: New Expensify. Device: Android SDK built for x86 - 10 - API 29. 
[conn] Failed to resolve new client Error: Unexpected null/undefined value found
Failed to find globally installed React DevTools: Error: ENOENT: no such file or directory, stat 'C:\Program Files\nodejs\node_modules\react-devtools\node_modules\react-devtools-core' 
Slow interaction 
"error" event intercepted: 
null
```

## To Reproduce

1. Develop an app locally on Android (e.g. emulator connected to metro server)
2. Launch flipper
3. Observe errors being logged 
4. Observe the app is not listed in flipper 
5. No plugins are being detected either (Hermes debugger works though)

## Environment
Windows 10
Android 10
Flipper  version 0.133.1 (same for 0.133.0) 
react-native: 0.66.4

**Comment by kidroca** [comment]
Reverting back to v0.128.4 fixes the issue for me 
![image](https://user-images.githubusercontent.com/12156624/153476063-4b5a61a5-0e27-4b9e-a73b-477cc5f92430.png)

**Comment by aigoncharov** [comment]
@kidroca could you click on "Log Levels", enable "info" logs, and share the complete logs?

**Comment by kidroca** [comment]
This is what I get with "info"

![image](https://user-images.githubusercontent.com/12156624/153498337-570e443b-aa59-4213-95d5-808c0761cbec.png)

```
[conn] Flipper server state -> starting 
[conn] Flipper server state -> pending 
8 plugin(s) were loaded in legacy mode. Please visit https://fbflipper.com/docs/extending/sandy-migration to learn how to migrate these plugins to the new Sandy architecture: 
Hermes Debugger (RN) (id: Hermesdebuggerrn)
KaiOS RAM graph (id: kaios-graphs)
KaiOS: big allocations (id: kaios-big-allocations)
Layout (id: Inspector)
LeakCanary (id: LeakCanary)
Performance (id: flipper-plugin-performance)
React Native Tic Tac Toe (id: ReactNativeTicTacToe)
Sandbox (id: Sandbox) 
[conn] secure server listening at port:  
8088
[conn] secure server (ws) listening at port:  
9088
[conn] insecure server listening at port:  
8089
[conn] insecure server (ws) listening at port:  
9089
[conn] Browser server (ws) listening at port:  
8333
falling back to the alternative adb path 
Failed to initialize ADB. Please disable Android support in settings, or configure a correct path. 
Error: spawn D:\Android\SDK\adb ENOENT
[conn] Flipper server state -> started 
Failed to find globally installed React DevTools: Error: ENOENT: no such file or directory, stat 'C:\Program Files\nodejs\node_modules\react-devtools\node_modules\react-devtools-core' 
[conn] RSockets are being deprecated at Flipper. Please, migrate to WebSockets. App: New Expensify. Device: Android SDK built for x86 - 10 - API 29. 
[conn] Secure rsocket connection attempt: New Expensify on emulator-5554. Medium 1. CSR: /data/user/0/[com.expensify.chat/files/sonar/](http://com.expensify.chat/files/sonar/) 
[conn] Connection established: New Expensify on emulator-5554. Medium 1. CSR: /data/user/0/[com.expensify.chat/files/sonar/](http://com.expensify.chat/files/sonar/) 
{sdk_version: 4, medium: 1, device_id: "emulator-5554", device: "Android SDK built for x86 - 10 - API 29", os: "Android"…}
[conn] Failed to resolve new client Error: Unexpected null/undefined value found
    at assertNotNull (file:///C:/Users/kidro/Desktop/FlipperWin/resources/app.asar/bundle.js:29:2352)

    at ServerController.addConnection (file:///C:/Users/kidro/Desktop/FlipperWin/resources/app.asar/bundle.js:7:6098)
4
Could not deliver message:  Error: Unexpected null/undefined value found
    at assertNotNull (file:///C:/Users/kidro/Desktop/FlipperWin/resources/app.asar/bundle.js:29:2352)

    at ServerController.addConnection (file:///C:/Users/kidro/Desktop/FlipperWin/resources/app.asar/bundle.js:7:6098)
```

**Comment by kidroca** [comment]
Hmm

```
Error: spawn D:\Android\SDK\adb ENOENT
```

The path to my adb is `D:\Android\SDK\platform-tools\adb.exe`
IDK why `platform-tools` is missing from the path in the log

**Comment by kidroca** [comment]
Seems like this path check returns `false` because of the `.exe` extension (maybe it's not a problem in unix?)

https://github.com/facebook/flipper/blob/4b6fcb6aaf886cb98c545e6f6f92ff478c07f7c7/desktop/flipper-server-core/src/devices/android/adbClient.tsx#L41-L44

**Comment by kidroca** [comment]
The change seems to fix the issue. Submitted a PR here: #3434 

IDK why a plugin I have is not appearing on neither enabled/disabled/unavailable lists. Could it be due to the local dev server? 
![image](https://user-images.githubusercontent.com/12156624/153607006-681204c0-9198-45f1-93d6-76506a14e3de.png)

It's not in the list of the detected plugins either
> ✅  Detected 18 bundled plugins: flipper-plugin-device-cpu, flipper-plugin-crash-reporter, flipper-plugin-databases, flipper-plugin-example, flipper-plugin-fresco, flipper-plugin-hermesdebuggerrn, flipper-plugin-kaios-big-allocations, flipper-plugin-kaios-graphs, flipper-plugin-inspector, flipper-plugin-leak-canary, flipper-plugin-device-logs, flipper-plugin-navigation, flipper-plugin-network, flipper-plugin-react-devtools, flipper-plugin-react-native-tic-tac-toe, flipper-plugin-sandbox, flipper-plugin-sea-mammals, flipper-plugin-preferences.

**Comment by aigoncharov** [comment]
@kidroca well, it is not a built-in plugin, is it? It explains why it is not listed as one of the bundled plugins. Yet, you should see it nevertheless. Would you mind creating a separate issue for that?

**Comment by aigoncharov** [comment]
Closed by #3434. Many thanks to @kidroca for providing the patch!
